### ISSUE TITLE
USWDS - Usability Testing: Combo box findings

### ISSUE THREAD

**Issue by mahoneycm** [uswds/uswds#5522]
## Description

Combo box findings, recommendations, and next steps from the [Aug-Sep 2023 Usability Testing Findings Report 🔒](https://docs.google.com/document/d/1z1BRxYZq_m65IdsRKsMzDYSpYYHKsqYgelQe7w_qDUQ/edit?usp=sharing)

[Testing video clips](https://drive.google.com/file/d/1h7TnVkejKvQGhPamrcZ5Z6rZxRrNl5A7/view?usp=drive_link) :lock:

## Findings takeaways

Pain points felt for 4 out of 5 participants:

- **Doesn’t behave as users expect**. Screen reader users, and possibly others, expect combo box to behave like a select. 
  - They expect items that start with the string they input to be at the beginning of the results. They expect to be able to use 1st letter navigation to get to their selection (“T” takes them to states starting with “T”) - [4 out of 5 participants]. One person said it acted as expected (searched using ‘Texas”) and this user seemed savvy/familiar with combobox functionality conventions.
  - Some users expect entering a repeated letter will step through options that start with that letter (i.e. from Tennessee to Texas) 
  - In general, the order of the results, and sometimes the results themselves, don't make sense immediately. 

- **The purpose of the clear button isn't obvious.** One person (U4) was surprised by the “x” button in field body. They saw the button as a deselect.  They usually only sees that in combo boxes where you can make multiple selections - [see clip](https://bixal.dovetailapp.com/data/7GYQdI9ZUR3yo3OuxEG2Oz#:v:h=7oEX7SwL4EKlIZYWp3vULf&s=1)

- Positive: One person liked that when you enter something wrong, it gives feedback of “no results found.” Often she sees combo boxes that give no feedback and are just blank.

There is past evidence/research for combobox that bolster the above findings:

- Combobox problems documented in this [combobox bug report](https://github.com/uswds/uswds/issues/5161). Perhaps the recommendation should be to use a SELECT component for state dropdowns, not a combobox.
- [Combobox feedback and recommendations](https://docs.google.com/document/d/11V_0QGD2NCNPLRJkH0gOkCw0kntY9Y_r8U33-hBd-yo/edit?usp=sharing) from research
    - *This document was from before a keyboard behavior enhancement in #5160  - Charlie*
- [Video clip of user struggling with first letter nav issue](https://drive.google.com/file/d/1J25wdVSczuRf7YUfvPpUdt6xH-TLQjvO/view?usp=sharing)

## 💡 Combo box original recommendations:

- Should recommendation be for state dropdowns to use the SELECT component instead of the COMBOBOX? We would need to change guidance if so, since current guidance says not to use select box for more than 15 options.
- Also, only offer the deselect “x” in comboboxes where the user can select multiple options.

### Note: Select vs Combobox

It appears the users are expecting behavior similar to a **Select** dropdown

From [W3C](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/#:~:text=Printable%20Characters%3A,the%20typed%20characters.):

>Printable Characters:
> - If the combobox is editable, type characters in the combobox. Note that some implementations may regard certain characters as invalid and prevent their input.
> - If the combobox is not editable, optionally moves focus to a value that starts with the typed characters.

In terms of W3C, our components operate as expected, but if users aren't expecting this behavior it could lead to confusion.

## Discussion

- Could we add screen reader instructions to enter a _search_ term?
- Combo box results: Could some of the unexpected behaviors be avoided by updating the combo box results to be sorted by relevance rather than alphabetically?
- combo box guidance improvements: How could we improve guidance to help devs properly implement this component?
- Finding 2: Deselect 'x' - Currently, this is the only way for users to deselect an option. If we get rid of it, there is no way to reset the combo box to it's initial state without reloading. Related PR - [Combobox: erase selection with backspace #4392](https://github.com/uswds/uswds/issues/4392)

**Comment by mejiaj** [comment]
Closing as completed with #6122.
